### PR TITLE
Views

### PR DESCRIPTION
**Pull request by olaishola05**
# Add views to the blog

- Implement the design from the wireframes design for the blog app
- Add view to display lists of users and number of posts on the home page
- Add a view for the individual user  with a display for recent posts
- Add a view to display all posts from a user
- Add a view to display a single post from a user
- Add minimal styling to all views

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,5 +1,10 @@
 class UsersController < ApplicationController
-  def index; end
+  def index
+    @users = User.all
+  end
```
Comment (Cathella) [inline app/controllers/users_controller.rb:4]: ✅ Users controller #index action renders all users to it's corresponding Views index page 👍🏻

File: app/views/users/_user.html.erb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<div class='user'>
+<div class='image-container'>
+  <img src=<%= user.photo %> >
+</div>
+  <div class='user-details'>
+    <h2><%= user.name %></h2>
+    <p>Number of posts: <%= user.posts.count %></p>
+  </div>
```
Comment (Cathella) [inline app/views/users/_user.html.erb:8]: ✅  Displays `user.photo`, `user.name` and `user.posts_counter` 👍🏻

File: app/views/posts/_post.html.erb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<div class='single-user-info'>
+    <h3>Post #<%= index + 1 %></h3>
+    <p><%= post.text%>...</p>
+    <div class='counters'>
+    <span>Comments: <%= post.comments.size ||= 'No comments yet' %></span>,
+    <span>Likes: <%= post.likes.size ||= 0 %></div></span>
```
Comment (Cathella) [inline app/views/posts/_post.html.erb]: - [ ] [REQUIRED] Kindly use the likes_counter -  `<small>Likes: <%= post.likes_counter || 0 %></small>`

File: app/views/users/show.html.erb [hunk 4]
```diff
@@ -1 +1,14 @@
...
+        <div class='single-user-info'>
+            <h3>Bio</h3>
+            <p><%= @user.bio %></p>
+        </div>
+        <h3 class='recent'>Recent Posts</h3>
+        <% @user.recent_posts.each_with_index do |post, index| %>
+            <%= render post, index: index %>
+        <% end %>
+    </div>
+    <button type='button' class='all-post-btn'>See all posts</button>
```
Comment (Cathella) [inline app/views/users/show.html.erb]: - [ ] [REQUIRED] The button or link should redirect to all posts path. See the suggestions below

`<%= link_to 'See all Post', user_posts_path(@user) %>`

File: app/views/posts/_post.html.erb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<div class='single-user-info'>
+    <h3>Post #<%= index + 1 %></h3>
+    <p><%= post.text%>...</p>
+    <div class='counters'>
+    <span>Comments: <%= post.comments.size ||= 'No comments yet' %></span>,
```
Comment (Cathella) [inline app/views/posts/_post.html.erb]: - [ ] [REQUIRED] Kindly use the recent_comments method to to render 5 most recent comments

```ruby
<% post.recent_comments.each do |comment| %>
  <p><%= comment.author.name %>: <%= comment.text %></p> 
 <% end %>
```

File: app/views/posts/show.html.erb [hunk 6]
```diff
@@ -1 +1,19 @@
-<h1>Show a single post</h1>
\ No newline at end of file
+<div class='post'>
+    <div class='single-post'>
+        <div class='post-header'>
+            <h3>Post #<%= @post.id %> by <%= @post.user.name %></h3>
+            <div class='counters post-counts'>
+                <span>Comments: <%= @post.comments.size ||= 'No comments yet' %></span>,
```
Comment (Cathella) [inline app/views/posts/show.html.erb]: - [ ] Kindly use the comments_counter instead

File: app/views/posts/show.html.erb [hunk 7]
```diff
@@ -1 +1,19 @@
...
+                <span>Likes: <%= @post.likes.size ||= 0 %></span>
```
Comment (Cathella) [inline app/views/posts/show.html.erb]: - [ ] Kindly use the likes _counter instead

**Review by Cathella — CHANGES_REQUESTED** [review]
## 🟡 Status: Changes Required 🛑

### Hi @netman5 👋🏻, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## To Highlight 👏🏻

✅ No linter errors.
✅ You used correct GitHub Flow.
✅ Descriptive pull request.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by bimbolabuari — APPROVED** [review]
Hi @netman5, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**